src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -149,26 +170,46 @@ public boolean requiresReference() {
     @Override
     public void onStartup() {
 
-        for (final String variantPath : variantFilePaths) {
+        List<VCFHeader> headers = new ArrayList<>(variantFilePaths.size());
 
-            final String variantURI = IOUtils.getPath(variantPath).toAbsolutePath().toUri().toString();
+        for (String variantFilePath : variantFilePaths) {
+
+            final String variantURI = IOUtils.getPath(variantFilePath).toAbsolutePath().toUri().toString();
             final Function<SeekableByteChannel, SeekableByteChannel> cloudWrapper = (cloudPrefetchBuffer > 0 ? is -> SeekableByteChannelPrefetcher.addPrefetcher(cloudPrefetchBuffer, is) : Function.identity());
             final Function<SeekableByteChannel, SeekableByteChannel> cloudIndexWrapper = (cloudIndexPrefetchBuffer > 0 ? is -> SeekableByteChannelPrefetcher.addPrefetcher(cloudIndexPrefetchBuffer, is) : Function.identity());
             final AbstractFeatureReader<VariantContext, LineIterator> reader = AbstractFeatureReader.getFeatureReader(variantURI, null, new VCFCodec(), true, cloudWrapper, cloudIndexWrapper);
 
+            VCFHeader header = (VCFHeader)reader.getHeader();
+
             // A GVCF file must contain only one sample, throw an exception otherwise
-            if (((VCFHeader)reader.getHeader()).getNGenotypeSamples() != 1) {
-                throw new UserException("Input GVCF: " + variantPath + " should contain data for one sample");
+            if (header.getNGenotypeSamples() != 1) {
+                throw new UserException("Input GVCF: " + variantFilePath + " should contain data for one sample");
+            }
+            String sampleName = header.getGenotypeSamples().get(0);
+            headers.add(header);
+            sampleNames.add(sampleName);
+            sampleNameToFileName.put(sampleName, variantFilePath);
+
+            // Import occurs in batches. By closing all readers here we release resources
+            // and preserve memory for later stages where readers are allocated
+            // for samples in a given batch
+            try {
+                reader.close();
+            } catch (IOException e) {
+                throw new GATKException("FeatureReader close() failed for " + sampleName, e);
             }
-
-            final String sampleName = ((VCFHeader) reader.getHeader()).getGenotypeSamples().get(0);
-            sampleToVCFMap.put(sampleName, reader);
         }
 
-        createMergedHeader();
+        mergedHeader = VCFUtils.smartMergeHeaders(headers, true);
 
-        initializeIntervals();
+        //This sorts the list sampleNames if !useSamplesInOrder
+        //Why you should use this? If you are writing multiple partitions in different machines,
+        //you must have consistent ordering of samples across partitions. If file order is different
+        //in different processes, then set useSamplesInOrder to false and let the sort in
+        //generateSortedCallSetMap ensure consistent ordering across samples
```

Comment:
I'd replace this long-ish comment with the more compact: "Passing in false here so that sample names will be sorted. This is needed for consistent ordering across partitions/machines."